src/lib/utils/ArtworkFiltersStore.tsx

```diff
@@ -0,0 +1,89 @@
+import { some, union } from "lodash"
+import React, { createContext, Dispatch, Reducer, useReducer } from "react"
+
+const filterState: ArtworkFilterContextState = {
+  appliedFilters: [],
+  selectedFilters: [],
+  selectedSortOption: "Default",
+}
+
+export const ArtworkFilterContext = createContext<ArtworkFilterContext>(null)
+
+export const ArtworkFilterGlobalStateProvider = ({ children }) => {
+  const [state, dispatch] = useReducer<Reducer<ArtworkFilterContextState, FilterActions>>(
+    (artworkFilterState, action) => {
+      switch (action.type) {
+        case "applyFilters":
+          const previouslyAppliedFilters = artworkFilterState.appliedFilters
+          const filtersToApply = action.payload
+          const appliedFilters = union(previouslyAppliedFilters, filtersToApply)
+          return { appliedFilters, selectedFilters: [], selectedSortOption: artworkFilterState.selectedSortOption }
+
+        case "selectFilters":
+          const previouslySelectedFilters = artworkFilterState.selectedFilters
+          const currentlySelectedFilter = action.payload
+          const isFilterAlreadySelected = some(previouslySelectedFilters, currentlySelectedFilter)
+          const selectedFilter: Array<{ type: SortOptions; filter: FilterOptions }> = []
+
+          if (isFilterAlreadySelected) {
+            const mergedFilters = previouslySelectedFilters.push(currentlySelectedFilter)
```

Comment:
We need to be careful here, reducers are not supposed to mutate the previous state because it could prevent react from applying updates correctly.

This should probably be 
```suggestion
            const mergedFilters = [...previouslySelectedFilters, currentlySelectedFilter];
```

To help avoid accidental mutation in the future we could update the `ArtworkFilterContextState` type to be something like this:

```ts
interface ArtworkFilterContextState {
  readonly appliedFilters: ReadonlyArray<{ readonly type: SortOptions; readonly filter: FilterOptions }>
  readonly selectedFilters: ReadonlyArray<{ readonly type: SortOptions; readonly filter: FilterOptions }>
  readonly selectedSortOption: SortOptions
}
```

It's a bit verbose I know 😬😅